flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -20,19 +29,31 @@ public class SwiftSentryFlutterPlugin: NSObject, FlutterPlugin {
     }
 
     private func registerObserver() {
+#if os(iOS)
         NotificationCenter.default.addObserver(self,
                                                selector: #selector(applicationDidBecomeActive),
                                                name: UIApplication.didBecomeActiveNotification,
                                                object: nil)
+#elseif os(macOS)
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(applicationDidBecomeActive),
+                                               name: NSApplication.didBecomeActiveNotification,
```

Comment:
@bruno-garcia a thing that I've noticed is that for macOS those notifications (within the Cocoa SDK) gets triggered when the App lost its focus but not actually when it gets minimized, is that the same behavior for Native macOS Apps?

Reply:
To be honest, I'm not sure. I believe it's still a fair approach. The other approach (taken by sentry-native) is to have a single session for the whole lifetime of the process. But how often do you close a desktop app these days? I only close it once a month to reboot.

Reply:
Sooo we'll leave it as is?

Reply:
yes, it was more an impl. detail question, this is handled by the Cocoa SDK anyway, so not related to this PR